### DISCUSSION TITLE
"Offline" usage

### DISCUSSION THREAD

**Discussion by nrace87** [niccokunzmann/open-web-calendar#456] [Q&A]
Hi, first of all I really like this project, I hope it is exatcly what I am looking for :) I like the "Try it out" thing a lot.
My idea is to let it run in a raspberry pi dashboard thing. I wondered if it is possible to store everything local, so create on this raspberry with an ics and the downloaded configuration a local website displaying the calendar. Specially as the ics may contain quite personal data. 

Thanks a lot :)

**Comment by niccokunzmann** [comment]
Hi, yes, the project is intended to work like that.
I personally use Python's `http.server` to run the tests and provide specifications and calendar ICS files.

Even if the Raspberry Pi is old, it should work with [Docker](https://open-web-calendar.quelltext.eu/host/docker/), too.
If you prefer the [Python installation](https://open-web-calendar.quelltext.eu/host/pypi/), add the [pip.conf](https://github.com/niccokunzmann/open-web-calendar/blob/74d62ef7f6939c3ec2b98b314ddc2738043a28d3/docker/pip.conf) file.

E.g. if you have these files in a folder `/home/pi/calendar/`:
- `specification.json` - the specification file
- `calendar.ics` - a calendar - TIP: you can export the ICS e.g. from Thunderbird to your local computer or phone and then sync this file with [syncthing](https://syncthing.net/) 

Then, I would add a start with `crontab -e`.

```
@reboot python3 -m http.server -d /home/pi/calendar/ -b localhost
```

After this, you can access the calendar and spec without the ICS being exposed to the network at [localhost:8000](http://localhost:8000).
E.g. if the open-web-calendar runs at `localhost:5000`:

```
http://localhost:5000/calendar.html?specification_url=http://localhost:8000/specification.json
```

`specification.json` - you can download a new one after editing the calendar (`?` at the bottom --> edit calendar)
```json
{
    "url" : "http://localhost:8000/calendar.ics"
}
```

You can also use other URLs and calendar sources. This works in a network without Internet access and synchronizes the calendar from your device securely as soon as you enter the network via syncthing.

Please let me know about your setup in the end! I am curious!
